crates/typst/src/image/raster.rs

```diff
@@ -7,12 +7,22 @@ use image::codecs::gif::GifDecoder;
 use image::codecs::jpeg::JpegDecoder;
 use image::codecs::png::PngDecoder;
 use image::io::Limits;
-use image::{guess_format, ImageDecoder, ImageResult};
+use image::{
+    guess_format, DynamicImage, GenericImageView, ImageDecoder, ImageResult, Rgba,
+};
+use once_cell::sync::{Lazy, OnceCell};
 use typst_macros::Cast;
 
 use crate::diag::{bail, StrResult};
 use crate::eval::Bytes;
 
+static THREAD_POOL: Lazy<rayon::ThreadPool> = Lazy::new(|| {
+    rayon::ThreadPoolBuilder::new()
+        .thread_name(|i| format!("typst-image-encode-{i}"))
```

Comment:
i guess the thread pool here will be used for all compute heavy tasks in future.

Reply:
I am now using the global one which solves this issue.